Fix Sidebar Update for Calamare's

Jkintalix opened the pull request:
- Replaced lambda with a real slot to satisfy Qt's UniqueConnection requirement. Taken from upstream commit c69e229

#### Merge Checklist  <!-- REQUIRED -->
**All** boxes should be checked before merging the PR
- [X] The changes in the PR have been built and tested
- [] cgmanifest file has been updated if required
- [X] Ready to merge

#### Description <!-- REQUIRED -->
Make Sidebar Update Properly in Calamares ViewManager

#### Any Newly Introduced Dependencies
No

#### How Has This Been Tested? 
Tested manually.

anujm1 commented:
> * Now the sidebar updates when moving through steps or sub-steps.
> * Added a slot to refresh the model and keep the sidebar in sync. 
> * Included direct updates and logs to help debug any issues

Please add this in patch file as well and submit this upstream as well.

Jkintalix committed:
Fix Sidebar Update

- Replaced lambda with a real slot to satisfy Qt's UniqueConnection requirement. Taken from upstream commit c69e229

Signed-off-by: kinatli jayanth <jayanthx.kintali@intel.com>
